src/App.js

```diff
@@ -1,16 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [chatMessagesData, setChatData] = useState(chatMessages);
+
+  const updateChatData = (updatedMessage) => {
+    const messages = chatMessagesData.map((message) => {
+      if (message.id === updatedMessage.id) {
+        return updatedMessage;
+      } else {
+        return message;
+      }
+
+    });
+    setChatData(messages);
+  };
+
+
+  const countLikes = (messages) => {
```

Comment:
This is a great way to use existing data. You could also use the reduce function
``` javascript
    return messageData.reduce((totalLikes, message) => {
      // If messages.liked is true add 1 to totalLikes, else add 0
      return (totalLikes += message.liked ? 1 : 0);
    }, 0); // The 0 here sets the initial value of totalLikes to 0
```